File: 0275. H-Index II/Solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int hIndex(vector<int>& citations)
    {
        int left = 0, right = citations.size() - 1, n = citations.size();
        while (left <= right) {
            int m = left + ((right - left) >> 1);
            if (citations[m] < n - m)
                left = m + 1;
            else
                right = m - 1;
        }
        return n - left;
    }
};

File: 0275. H-Index II/Solution.py
from typing import List


class Solution:
    def hIndex(self, citations: List[int]) -> int:
        n = len(citations)
        left, right = 0, n - 1
        while left <= right:
            m = left + (right - left) // 2
            if citations[m] < n - m:
                left = m + 1
            else:
                right = m - 1
        return n - left
